crates/turbo-tasks/src/state.rs

```diff
@@ -133,10 +155,35 @@ impl<T: PartialEq> State<T> {
     }
 }
 
-impl<'a, T> std::ops::Deref for StateRef<'a, T> {
+impl<'a, T> Deref for StateRef<'a, T> {
```

Comment:
Nit: Now that only `deref_mut` will invalidate the value, I think we can remove `StateRef` entirely (and possibly rename `StateRefMut` to `StateRef`).